README.md

```diff
@@ -501,54 +543,82 @@ To create a dynamic list of counter we have to use the `list` function.
 
 ### Understanding generator functions
 
-At first Turbine's use of generator functions may seem peculiar. But
-it is just a nicer way to call [`chain`](#componentchain) methods.
+Turbine's use of generator functions may seem a bit puzzling at first.
+For instance, it may seem like generator functions serve two different
+purposes. One when they're used in the model and another when they're
+used in the view
 
-Three types in Turbine have a `chain`-method: `Component`, `Now` and
-`Behavior`. `chain` is used to compose structures together such that the result
-from one structure can be used in the next.
+But, what they do under the hood is exactly the same in both cases.
+The key to understand is that generator functions is just sugar for
+calling [`chain`](#componentchain) several times in succession.
 
-For instance, when we use `chain` on components we can combine
-elements as we'd like and pipe output from one component into the
-next. The code below combines two `input` elements with a `span`
-element that shows the concatenation of the text in the two input
-fields.
+When we use `chain` on components we can combine elements and pipe
+output from one component into the next. The code below combines two
+`input` elements with a `span` element that shows the concatenation of
+the text in the two input fields.
 
 ```typescript
 input({ attrs: { placeholder: "foo" } }).chain(
-  ({ inputValue: a }) => input().chain(
-    ({ inputValue: b }) => span(["Combined text: ", a, b])
+  ({ inputValue: aValue }) => input().chain(
+    ({ inputValue: bValue }) => {
+      const concatenated = lift((a, b) => a + b, aValue, bValue);
+      return span(["Concatenated text: ", concatenated]).mapTo({concatenated});
+    }
   )
 );
 ```
 
 However, the above code is very awkward as each invocation of `chain`
-adds an extra layer of nesting. To solve the problem we use
+adds an extra layer of nesting. To solve this problem we use
 generators.
 
 ```typescript
 go(function*() {
-  const {inputValue: a} = yield input();
-  const {inputValue: b} = yield input();
-  yield span(["Combined text: ", a, b]);
+  const {inputValue: aValue} = yield input();
+  const {inputValue: bValue} = yield input();
+  const concatenated = lift((a, b) => a + b, aValue, bValue);
+  yield span(["Concatenated text: ", concatenated]);
+  return {concatenated};
 });
 ```
 
 The above code does exactly the same as the previous example. But it
 is a lot easier to read!
 
 The `go` function works like this. We yield a value with a `chain`
-method. `go` then calls `chain` on the yielded value. The function
-that `go` gives to `chain` continues the generator function with the
+method. `go` then calls `chain` on the yielded value. `go` calls
+`chain` with a function that continues the generator function with the
 value that `chain` passes it. The end result is a value of the same
```

Comment:
Maybe avoid saying "function continues", because this is not the native feature of generators, but rather the specific interpretation the `go` puts on them?

Maybe something like "the function taking scope variables and returning the next yielded expression"?